### ISSUE TITLE
Initial DataStore Sync Slow for Android Devices

### ISSUE THREAD

**Issue by jackcho0301** [aws-amplify/amplify-flutter#4686]
Hello, 

My development team is having trouble with initial syncing using DataStore, specifically on Android. 

### Description of the issue:
After a fresh install of the app, and once a user signs in using Amplify's Authenticator widget, a user should be able to have access to models fetched from DataStore. We use `Amplify.DataStore.observeQuery()` to fetch this specific model.

On iOS devices, the initial sync for DataStore seems to happen fast and `Amplify.DataStore.observeQuery()` fetches snapshots instantly. But it takes a long time for Android devices to sync data for the first time, resulting in `observeQuery()` fetching empty snapshots a multiple times, and finally fetching a complete snapshot.

I noticed that observeQuery() on Android will fetch the complete snapshot once the following logs are shown on the terminal: 

```
I/amplify:aws-datastore(12906): Successfully sync'd down model state from cloud.
I/amplify:aws-datastore(12906): Starting processing subscription data buffer.
I/amplify:aws-datastore(12906): Started the orchestrator in API sync mode.
I/amplify:aws-datastore(12906): Started processing the mutation outbox. Pending mutations will be published to the cloud.
```

So my team reached a conclusion that DataStore takes long time for syncing the models. And we need your guidance to help resolve this issue. This has been an ongoing issue for almost a year, and we really hope to understand why initial DataStore sync is slow for Android devices. We can provide an obfuscated version of our code and/or GraphQL schema if it would help. 

Also, are there best practices that would help data models to be synced faster? Our team was considering reducing the number of models being synced by using DataStoreSyncExpression to selectively sync certain models only. And we could choose to not sync some models that do not benefit from DataStore's local-first behavior in our use case. If this approach does not sound reasonable, please let us know what the proper best practices would be.

### Misc information: 
Flutter Version: 3.10.5
amplify_flutter Version: 1.4.1
amplify_datastore Version: 1.4.2
compileSdkVersion on android/app/build.gradle: 33

Thank you for your time!

**Comment by Equartey** [comment]
Hi @jackcho0301, This is something our team is aware of and currently working to address. We'll update your more concretely when a fix is released. 

In the meantime, can you confirm the Android version you're observing the issue on? And the amount of models your project has?

**Comment by jackcho0301** [comment]
Hello @Equartey , thank you for the response. We are observing this issue on Android 14, both physical and virtual devices. And we have 12 models for our project.

**Comment by amitchaudhary140** [comment]
Yes I am also facing same issue on Android.

**Comment by NikaHsn** [comment]
Thank you for your patience. The issue has been fixed in version 1.8.0. and I'm closing it. However, if you encounter any issues after updating to version 1.8.0, please don't hesitate to reopen it.
